#!/usr/bin/env python
PACKAGE = "spc_uav_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mass",   	double_t, 0, "Mass",		  		              		  1.77, 0.00,  7.00)
gen.add("J_xx", 	double_t, 0, "Inertia Matrix xx", 	        			  0.0347563, 0.00,  0.9)
gen.add("J_yy", 	double_t, 0, "Inertia Matrix yy", 	        			  0.0458929, 0.00,  0.9)
gen.add("J_zz", 	double_t, 0, "Inertia Matrix zz", 	        			  0.0977, 0.00,  0.9)

gen.add("K_v_x", 	double_t, 0, "Velocity Gain - x", 	        			  4.70, 0.00,  100.00)
gen.add("K_v_y", 	double_t, 0, "Velocity Gain - y", 	        			  4.70, 0.00,  100.00)
gen.add("K_v_z", 	double_t, 0, "Velocity Gain - z", 	        			  4.70, 0.00,  100.00)

gen.add("K_w_x", 	double_t, 0, "Angular Rate Gain - x", 				      0.5, 0.00,  10.00)
gen.add("K_w_y", 	double_t, 0, "Angular Rate Gain - y", 				      0.5, 0.00,  10.00)
gen.add("K_w_z", 	double_t, 0, "Angular Rate Gain - z", 				      0.5, 0.00,  10.00)
gen.add("K_w_x_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - x", 		  0.2, 0.00,  1.00)
gen.add("K_w_y_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - y", 		  0.2, 0.00,  1.00)
gen.add("K_w_z_px4",double_t, 0, "PX4 Onboard Angular Rate Gain - z", 		  0.3, 0.00,  1.00)
gen.add("c_KP_r_xx",double_t, 0, "Admittence filter Spring K_Pr Matrix xx",   3.0, 0.0,  20.0)
gen.add("c_KP_r_yy",double_t, 0, "Admittence filter Spring K_Pr Matrix yy",   3.0, 0.0,  20.0)
gen.add("c_KP_r_zz",double_t, 0, "Admittence filter Spring K_Pr Matrix zz",   3.0, 0.0,  20.0)
gen.add("c_KP_t_xx",double_t, 0, "Admittence filter Spring K_Pt Matrix xx",   3.0, 0.0,  25.0)
gen.add("c_KP_t_yy",double_t, 0, "Admittence filter Spring K_Pt Matrix yy",   3.0, 0.0,  25.0)
gen.add("c_KP_t_zz",double_t, 0, "Admittence filter Spring K_Pt Matrix zz",   3.0, 0.0,  25.0)
gen.add("track_end_effector", 	bool_t, 0, "End Effector Tracking Switch",  False)
exit(gen.generate(PACKAGE, "spc_uav_control", "Impedance_Control_Gains"))
